python/ray/tune/impl/tuner_internal.py

```diff
@@ -191,8 +198,77 @@ def _maybe_warn_resource_contention(self):
                 stacklevel=4,
             )
 
+    def _validate_overwrite_trainable(self, original_trainable, overwrite_trainable):
+        """Determines whether the re-specified overwrite_trainable is compatible
+        the restored experiment with some basic sanity checks
+        (ensuring same type and name as the original trainable).
+        """
+
+        # Check if the trainable was wrapped with `tune.with_parameters`,
+        # Set the Tuner to fail on fit if the trainable is not re-specified.
+        trainable_wrapped_params = getattr(
+            original_trainable, "_attached_param_names", None
+        )
+        if trainable_wrapped_params and not overwrite_trainable:
+            self._missing_params_error_message = (
```

Comment:
Would it be okay to just save the missing param names like I did in the other PR, then construct the error message in `_fit_resume`?

Reply:
is it possible for `_fit_resume` to be called without resetting the `self._missing_params_error_message` first? I want to avoid a situation where the state changes to one that allows for `_fit_resume` to progress but the attribute is not updated, leading to an error being raised again.

Reply:
I see, I think there's no way to call `_restore_from_path_or_uri` again from the external `Tuner` API, so the `self._missing_params_error_message` will always reflect the correct state at `_fit_resume`. I do understand the concern though - having the check/error in happen different places is adding complexity.

Reply:
Exactly - that being said, if there's only one code path, it should be fine here